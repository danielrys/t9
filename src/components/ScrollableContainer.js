// @flow
import * as React from "react"
import { View, StyleSheet, ScrollView } from "react-native"

// components
import Container from "./Container"

import type { ViewType } from "../types"

type ScrollableContainerProps = {
  children: React.Node,
  style?: ViewType,
  contentStyle?: ViewType,
  centerContent?: boolean,
  loading: boolean,
}

const styles = StyleSheet.create({
  centeredWrapper: {
    flex: 1,
  },
  centeredContent: {
    flexGrow: 1,
    justifyContent: "center",
  },
})

export default class ScrollableContainer extends React.PureComponent<ScrollableContainerProps> {
  static defaultProps = {
    style: {},
    contentStyle: {},
    centerContent: false,
  }

  render() {
    const { children, style, loading, centerContent, contentStyle } = this.props
    const innerStyles = centerContent ? [styles.centeredContent] : []
    const wrapperStyles = centerContent ? [styles.centeredWrapper] : []

    return (
      <View style={[...wrapperStyles, style]}>
        {loading ? (
          <Container loading />
        ) : (
          <ScrollView
            showsVerticalScrollIndicator={false}
            contentContainerStyle={[...innerStyles, contentStyle]}
          >
            {children}
          </ScrollView>
        )}
      </View>
    )
  }
}
