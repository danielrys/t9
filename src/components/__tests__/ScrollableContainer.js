import React from "react"
import renderer from "react-test-renderer"

import { Text } from "react-native"

// components
import ScrollableContainer from "../ScrollableContainer"

describe("ScrollableContainer component", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(
        <ScrollableContainer>
          <Text>Hi</Text>
        </ScrollableContainer>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("renders with a centered content", () => {
    const tree = renderer
      .create(
        <ScrollableContainer centerContent>
          <Text>Hi</Text>
        </ScrollableContainer>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
