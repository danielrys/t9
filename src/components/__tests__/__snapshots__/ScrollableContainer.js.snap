// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableContainer component renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
      ]
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Text>
        Hi
      </Text>
    </View>
  </RCTScrollView>
</View>
`;

exports[`ScrollableContainer component renders with a centered content 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {},
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "flexGrow": 1,
          "justifyContent": "center",
        },
        Object {},
      ]
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Text>
        Hi
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
